text/0118-search-api.md

```diff
@@ -108,156 +109,344 @@ It returns the number of documents matching the current search query for each gi
 
-##### 1.2.1.7 `attributesToRetrieve`
+#### 3.1.7. `attributesToRetrieve`
 
 - Type: Array of String (POST) | String (GET)
 - Required: False
 - Default: `[]|null`
 
 Configures which attributes will be retrieved in the returned documents.
 
-If no value is specified, `attributesToRetrieve` uses the `displayedAttributes` list, which by default contains all attributes found in the documents.
+If no value is specified, `attributesToRetrieve` uses the `displayedAttributes` index setting, which by default contains all attributes found in the documents.
 
-> If an attribute has been removed from `displayedAttributes` index settings, `attributesToRetrieve` will silently ignore it and the field will not appear in the returned documents.
+> If an attribute is missing from `displayedAttributes` index setting, `attributesToRetrieve` silently ignore it, and the field doesn't appear in the returned search results.
 
-- 🔴 Sending a value with a different type than `Array of String`(POST), `String`(GET) or `null` for `attributesToRetrieve` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `Array of String`(POST), `String`(GET) or `null` for `attributesToRetrieve` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
 
-##### 1.2.1.8 `attributesToHighlight`
+#### 3.1.8. `attributesToHighlight`
 
-- Type: Array[String](POST)|String(GET)
+- Type: Array of String (POST) | String(GET)
 - Required: False
 - Default: `[]|null`
 
-Highlights matching query terms in the specified attributes by enclosing them in `<em>` tags.
+Highlights terms in the documents that correspond to the query terms (i.e. the terms in the [`q`](#311-q) search parameter) for the specified attributes.
+
+Search results include a `_formatted` object containing the highlighted terms when this parameter is defined. See [3.2.1.1.2. `_formatted`](#32112-formatted) section.
+
+Highlighted parts are surrounded by the [`highlightPreTag`](#319-highlightpretag) and [`highlightPostTag`](#3110-highlightposttag) parameters.
+
+If `"*"` is provided as a value: e.g. `"attributesToHighlight":["*"]`, all the attributes present in `attributesToRetrieve` will be assigned to `attributesToHighlight`.
+
+`attributesToHighlight` only works on values of the following types: `string`, `number`, `array`, `object`. When highlighted, number attributes are transformed to string.
+
+- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToHighlight` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 3.1.8.1. searchableAttributes
+
+Attributes not defined in the `searchableAttributes` index setting are also highlighted if assigned to `attributesToHighlight`.
+
+##### 3.1.8.2. stopWords
+
+Attributes defined in the `stopWords` index setting are also highlighted if matched.
```

Comment:
Yes, I don't think it is a problem to highlight the terms written in the query terms whether they are stop-words or not.

The formatting of the search results is there to help the reading of the search results by highlighting the query terms contained in a search result. If a query term typed by the user can be highlighted, it should help the user read the results.

Stop-words are used to avoid degrading relevancy when they populate an attribute densely. I'm currently not convinced that removing them helps more than anything else regarding the reading of search results.

If you have arguments that help me to understand why it does not improve the reading of the search results if they are there, I am ready to reconsider the question!  

Good point about synonyms, I didn't know about this current behavior, I will specify it because it's nice.

Reply:
Your thoughts are interesting, I think these are topics that can be addressed later if they need to be. Here is my thought to give more context:

- It describes the current behavior of the engine (for stop-words and synonyms). If this was a problem for a user at one time, it is not important enough to be important now. Changing this behavior could impact other users, it would involve adding an option to allow choosing but would add complexity and time to hit the v0.27.0 while we want to mostly reduce pain points regarding the customizable highlight tag / poor cropper that forces users to use `_matchesInfo` to do this work on the frontend-side.
- I also checked this behavior which is present by default in Algolia. With so little feedback and this behavior already in place, I think we can base our choices on what a well-placed and well-rated actor does.

I can give my feedback that goes in the opposite direction of your idea about stop words to give the change but again it's just to confront it.

If a query contains several stop-words, e.g. `The son was on the path of redemption`.  In this case, doesn't removing the stop-words complicate the readability if the exact phrase can be highlighted?